zeeguu/core/word_scheduling/basicSR/basicSR.py

```diff
@@ -116,9 +143,63 @@ def find_or_create(cls, db_session, bookmark):
         db_session.commit()
         return b
 
+    @classmethod
+    def priority_bookmarks_to_study(cls, user, required_count):
+        """
+        Prioritizes the bookmarks to study. To randomize the
+        exercise order utilize the Frontend assignBookmarksToExercises.js
+
+        The original logic is kept in bookmarks_to_study as it is called to
+        get similar_words to function as distractors in the exercises.
+
+        Currently, we prioritize bookmarks in the following way:
+        1. Words that are closest to being learned (indicated by `cooling_interval`, the highest the closest it is)
+        2. Words that are most common in the language (utilizing the word rank in the db)
+        """
+
+        def sorting_properties(bookmark):
+            cooling_interval = (
+                cls.query.filter_by(bookmark=bookmark).one().cooling_interval
```

Comment:
But, we already have the bookmark object loaded in memory. We should be able to simply say: 

```
bookmark.cooling_interval
```

Reply:
I tried this, but I get an error: 

2024-02-21 17:12:15   File "/Zeeguu-API/zeeguu/core/word_scheduling/basicSR/basicSR.py", line 161, in sorting_properties
2024-02-21 17:12:15     cooling_interval = bookmark.cooling_interval
2024-02-21 17:12:15 AttributeError: 'Bookmark' object has no attribute 'cooling_interval'

I don't think it actually stores it from the query, or should I do something else?

Reply:
sorry, this is stupid. Indeed, the cooling interval is in the BasicSR object, not in the bookmark.